### PR TITLE
fix(ArtificialRotator): Add rotate coroutine cancel before new rotation

### PR DESCRIPTION
**Pull request by ShidoArima**
Fix for SetAngleTarget bug caused by still working coroutine.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2a1be5b6c96791222e2fa1ca585a9ff2a75366bd
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/thestonefox/SteamVR_Unity_Toolkit

---
This message was auto-generated by https://gitcop.com

**Comment by kristianpd** [comment]
You'll need to update the commit message to get this approved: 

> fix(ArtificialRotator): always cancel update rotation in SetRotation

?

File: Assets/VRTK/Source/Scripts/Interactions/Interactables/GrabAttachMechanics/VRTK_RotateTransformGrabAttach.cs [hunk 1]
```diff
@@ -278,6 +278,7 @@ public virtual void SetRotation(float newAngle, float transitionTime = 0f)
             }
             else
             {
+				CancelUpdateRotation();
```
Comment (kristianpd) [inline Assets/VRTK/Source/Scripts/Interactions/Interactables/GrabAttachMechanics/VRTK_RotateTransformGrabAttach.cs]: You should move this to https://github.com/thestonefox/VRTK/pull/1774/files#diff-2c617bc088c6541ec6c8322f2c6c5915R273 and take it out of the `true` condition as well.

also, you are missing an extra space indentation here

Comment (ShidoArima) [reply Assets/VRTK/Source/Scripts/Interactions/Interactables/GrabAttachMechanics/VRTK_RotateTransformGrabAttach.cs]: Good point. Ok

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2a1be5b6c96791222e2fa1ca585a9ff2a75366bd
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/thestonefox/SteamVR_Unity_Toolkit

---
This message was auto-generated by https://gitcop.com

File: Assets/VRTK/Source/Scripts/Interactions/Interactables/GrabAttachMechanics/VRTK_RotateTransformGrabAttach.cs [hunk 2]
```diff
@@ -271,16 +271,16 @@ public virtual void SetRotation(float newAngle, float transitionTime = 0f)
...
 
+            CancelUpdateRotation();
             if (transitionTime > 0f)
             {
-                CancelUpdateRotation();
                 updateRotationRoutine = StartCoroutine(RotateToAngle(newCurrentRotation, VRTK_SharedMethods.DividerToMultiplier(transitionTime)));
             }
             else
-            {
+            {
```
Comment (kristianpd) [inline Assets/VRTK/Source/Scripts/Interactions/Interactables/GrabAttachMechanics/VRTK_RotateTransformGrabAttach.cs]: can you delete the whitespace here please

**Comment by kristianpd** [comment]
Once you're done you'll want to `git rebase -i master` your commits and delete the old one. If you `ammend` a commit without force pushing the changes, it will show up as a new commit.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 38473d756f4135997dd110d70b2561a45d4b87e9
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/thestonefox/SteamVR_Unity_Toolkit

---
This message was auto-generated by https://gitcop.com

**Comment by ShidoArima** [comment]
@kristianpd something goes wrong, now after rebase i see your commit too.

**Comment by thestonefox** [comment]
This PR needs tidying up, too many commits

**Comment by thestonefox** [comment]
No longer merge-able with the new master branch. Closing.

**Pull request closed without merging**